src/python/pants/backend/kotlin/test/junit.py

```diff
@@ -0,0 +1,84 @@
+# Copyright 2022 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+from __future__ import annotations
+
+from dataclasses import dataclass
+
+from pants.backend.kotlin.subsystems.kotlin import KotlinSubsystem
+from pants.backend.kotlin.target_types import KotlinJunitTestDependenciesField
+from pants.build_graph.address import Address
+from pants.engine.internals.selectors import Get
+from pants.engine.rules import collect_rules, rule
+from pants.engine.target import InjectDependenciesRequest, InjectedDependencies, WrappedTarget
+from pants.engine.unions import UnionRule
+from pants.jvm.dependency_inference.artifact_mapper import (
+    AllJvmArtifactTargets,
+    find_jvm_artifacts_or_raise,
+)
+from pants.jvm.resolve.common import Coordinate
+from pants.jvm.subsystems import JvmSubsystem
+from pants.jvm.target_types import JvmResolveField
+
+
+class InjectKotlinJunitTestDependencyRequest(InjectDependenciesRequest):
+    inject_for = KotlinJunitTestDependenciesField
+
+
+@dataclass(frozen=True)
+class KotlinJunitLibrariesForResolveRequest:
+    resolve_name: str
+
+
+@dataclass(frozen=True)
+class KotlinJunitLibrariesForResolve:
+    addresses: frozenset[Address]
```

Comment:
Yay, thanks for using `frozenset` rather than `FrozenOrderedSet`: https://github.com/pantsbuild/pants/issues/14719#issuecomment-1122533481